numeric.c

```diff
@@ -5739,24 +5739,60 @@ int_round(int argc, VALUE* argv, VALUE num)
 }
 
 /*
+ *  :markup: markdown
+ *
  *  call-seq:
  *    floor(ndigits = 0) -> integer
  *
- *  Returns the largest number less than or equal to +self+ with
- *  a precision of +ndigits+ decimal digits.
+ *  Returns an integer that is a "floor" value for `self`,
+ *  as specified by the given `ndigits`,
+ *  which must be an
+ *  [integer-convertible object](rdoc-ref:implicit_conversion.rdoc@Integer-Convertible+Objects).
  *
- *  When +ndigits+ is negative, the returned value
- *  has at least <tt>ndigits.abs</tt> trailing zeros:
+ *  - When `self` is zero, returns zero (regardless of the value of `ndigits`):
  *
- *    555.floor(-1)  # => 550
- *    555.floor(-2)  # => 500
- *    -555.floor(-2) # => -600
- *    555.floor(-3)  # => 0
+ *      ```
+ *      0.floor(2)  # => 0
+ *      0.floor(-2) # => 0
+ *      ```
  *
- *  Returns +self+ when +ndigits+ is zero or positive.
+ *  - When `self` is non-zero and `ndigits` is non-negative, returns `self`:
+ *
+ *      ```
+ *      555.floor     # => 555
+ *      555.floor(50) # => 555
+ *      ```
+ *
+ *  - When `self` is non-zero and `ndigits` is negative,
+ *    returns a value based on a computed granularity:
+ *
+ *      - The granularity is <tt>ndigits.abs * 10</tt>.
+ *      - The returned value is the largest multiple of the granularity
+ *        that is less than or equal to `self`.
+ *
+ *      Examples with positive `self`:
+ *
+ *      | ndigits | Granularity | 1234.floor(ndigits) |
+ *      |--------:|------------:|--------------------:|
+ *      | -1      | 10          | 1230                |
+ *      | -2      | 100         | 1200                |
+ *      | -3      | 1000        | 1000                |
+ *      | -4      | 10000       | 0                   |
+ *      | -5      | 100000      | 0                   |
+ *
+ *      <br>
+ *
+ *      Examples with negative `self`:
+ *
+ *      | ndigits | Granularity | -1234.floor(ndigits) |
+ *      |--------:|------------:|---------------------:|
+ *      | -1      | 10          | -1240                |
+ *      | -2      | 100         | -1300                |
+ *      | -3      | 1000        | -2000                |
+ *      | -4      | 10000       | -10000               |
+ *      | -5      | 100000      | -100000              |
  *
- *    555.floor     # => 555
- *    555.floor(50) # => 555
+ *      <br>
```

Comment:
Do you think we'd be able to avoid these? They pollute the markup displayed inside `ri`.

Markdown supports a line break using two or more white-space characters at the end of the line. Perhaps we could try that?

Reply:
Thanks, @adam12.  I'll look into it.  (There are others, elsewhere.)

Reply:
@adam12, I've removed the offending raw HTML.  The trailing whitespace did not work for me in rdoc 6.6.2 -- did not generate a line break -- so I have not added it in the recent commit.

Reply:
Thanks for trying. I wonder if this is a limitation in the markdown parser that ships inside RDoc. I know the feature itself isn't well documented.

Reply:
Looks like Markdown parser handles it correctly but maybe ignored when transforming the AST.

```
$ ruby -rrdoc -e '
pp RDoc::Markdown.parse(<<EOM)
First line with two spaces after.  
And the next line.
EOM'
[doc: [para: "First line with two spaces after.", [break], "And the next line."]]
```

Reply:
So what does that tell us (relative to this PR)?

Reply:
You've already removed the HTML markup so I'm presuming the extra linebreak was just for aesthetics and not critical. If that's the case, I'm just musing in our existing discussion and don't think it's holding anything up.

Reply:
Thanks, @adam12.

Reply:
Closing the loop on this, RDoc + Markdown where there are two spaces trailing a line will indeed insert a linebreak, but only if the line has non-whitespace characters. Empty lines or lines with just whitespace are collapsed.

Reply:
I thought I tried putting two spaces after the vertical bar in the table, but maybe I didn't.   Will try again when I remove the remaining `<br>`s from other markdown passages.